lib/commands/interactive.js

```diff
@@ -34,52 +32,55 @@ function getReplParameters(opts, suitest) {
 	return [vars, cwd, watch, file];
 }
 
-/**
- * Start interactive REPL
- * @param {String=} opts.cwd
- * @param {Array|String=} opts.watch
- * @param {Array|String=} opts.ignored
- * @param {Object=} opts.vars
- * @param {Function=} opts.repeater
- * @returns {Promise} response object
- */
-async function interactive(opts = {}) {
-	if (semver.lt(process.version, '9.4.0')) {
-		logger.error(texts.replWrongNodeVersion(process.version));
-		process.exit(1);
-	}
+const interactiveFactory = (classInstance) => {
+	/**
+	 * Start interactive REPL
+	 * @param {String=} opts.cwd
+	 * @param {Array|String=} opts.watch
+	 * @param {Array|String=} opts.ignored
+	 * @param {Object=} opts.vars
+	 * @param {Function=} opts.repeater
+	 * @returns {Promise} response object
+	 */
+	return async function interactive(opts = {}) {
+		const {logger, authContext} = classInstance;
+		if (semver.lt(process.version, '9.4.0')) {
+			logger.error(texts.replWrongNodeVersion(process.version));
+			process.exit(1);
+		}
 
-	if (isReplActive)
-		return Promise.resolve();
+		if (isReplActive)
+			return Promise.resolve();
 
-	const {replWarnInteractive, replWelcomeMessage, replSessionEnded} = texts;
+		const {replWarnInteractive, replWelcomeMessage, replSessionEnded} = texts;
 
-	if (!authContext.isInteractiveSession()) {
-		logger.warn(replWarnInteractive());
+		if (!authContext.isInteractiveSession()) {
+			logger.warn(replWarnInteractive());
 
-		return Promise.resolve();
-	}
+			return Promise.resolve();
+		}
 
-	const [vars, cwd, watch, callFile] = getReplParameters(opts, this);
-	const repeater = getShortFunctionBody(opts.repeater);
+		const [vars, cwd, watch, callFile] = getReplParameters(opts, this);
+		const repeater = getShortFunctionBody(opts.repeater);
 
-	logger.special(replWelcomeMessage(
-		vars, cwd, watch, repeater, logger.colors.suit
-	));
+		logger.special(replWelcomeMessage(
+			vars, cwd, watch, repeater, logger.colors.suit
+		));
 
-	isReplActive = true;
+		isReplActive = true;
 
-	await repl.startRepl({
-		cwd,
-		vars,
-		watch,
-		callFile,
-		ignored: opts.ignored,
-		repeater: opts.repeater,
-	});
+		await repl.startRepl({
+			cwd,
+			vars,
+			watch,
+			callFile,
+			ignored: opts.ignored,
+			repeater: opts.repeater,
+		});
 
-	isReplActive = false;
-	logger.info(replSessionEnded());
-}
+		isReplActive = false;
+		logger.info(replSessionEnded());
+	};
+};
 
-module.exports = interactive;
+module.exports = interactiveFactory;
```

Comment:
I would leave this file a singleton and require it dynamically only when it's actually needed (i.e. we have interactive session started through launcher's child process).